waddrmgr/db.go

```diff
@@ -1865,3 +1816,165 @@ func upgradeToVersion3(namespace walletdb.Namespace, seed, privPassPhrase, pubPa
 	}
 	return nil
 }
+
+// upgradeToVersion4 upgrades the database from version 3 to version 4
+// The following fields were added to the serialized address row:
+// * 'watchingOnly' - boolean indicating whether the address is watching-only
+// This is required to flag each imported address as watching-only
+// * 'encryptedPubKeyHash' - encrypted pubkey hash of imported p2pkh address
+// This is required to import addresses without the associated public key.
+func upgradeToVersion4(namespace walletdb.Namespace) error {
+	err := namespace.Update(func(tx walletdb.Tx) error {
+		currentMgrVersion := uint32(4)
+		bucket := tx.RootBucket().Bucket(addrBucketName)
+
+		err := bucket.ForEach(func(k, v []byte) error {
+			// Skip buckets.
+			if v == nil {
+				return nil
+			}
+
+			// Insert watchingOnly bool at byte 14
+			// Since importing watching-only addrs was added in this version we
```

Comment:
Minor nit: I've made it a point to avoid personal pronouns in the address manager.  Please change this to:

```Go
			// Since importing watching-only addrs was not added
			// until this version, mark all existing addresses as
			// not being watching-only.
```